File: src/main/scala/org/xbery/artbeams/categories/service/CategoryService.scala
package org.xbery.artbeams.categories.service

import org.xbery.artbeams.categories.domain.{Category, EditedCategory}
import org.xbery.artbeams.common.context.OperationCtx

/**
  * @author Radek Beran
  */
trait CategoryService {
  def findCategories(): Seq[Category]

  def saveCategory(edited: EditedCategory)(implicit ctx: OperationCtx): Either[Exception, Option[Category]]

  def findBySlug(slug: String): Option[Category]
}

File: src/main/kotlin/org/xbery/artbeams/categories/service/CategoryService.kt
package org.xbery.artbeams.categories.service

import org.xbery.artbeams.categories.domain.Category
import org.xbery.artbeams.categories.domain.EditedCategory
import org.xbery.artbeams.common.context.OperationCtx

/**
 * @author Radek Beran
 */
interface CategoryService {
    fun findCategories(): List<Category>
    fun findCategoriesByArticleId(articleId: String): List<Category>
    fun saveCategory(edited: EditedCategory, ctx: OperationCtx): Category?
    fun findBySlug(slug: String): Category?
}
